NIPTool/exeptions.py

```diff
@@ -0,0 +1,7 @@
+
+class MissingResultsError(Exception):
+    pass
+
+
+class FileValidationError(Exception):
+    pass
```

Comment:
```suggestion


class NIPToolError(Exception):
    def __init__(self, message):
        self.message = message

class MissingResultsError(NIPToolError):
    pass


class FileValidationError(NIPToolError):
    pass
```

Reply:
I think its good to create a base error class for all tool-related errors!
Now all the NIPTool errors will be of the same base class - and can be excepted together when NIPToolError is excepted.
Adding message to init also allows you to pass messages when error is raised:
`  raise MissingResultsError("No results found")`

When catching all errors downstream in the CLI, you can then simply do:
```
except NIPToolError as e:
    LOG.error(e.message)
```

to gather all errors and messages into one go

Reply:
Super! I learn so much from this! Thanks